examples/ann/annoy_nsmlib_example.ipynb

```diff
@@ -0,0 +1,725 @@
+      "text/plain": [
+       "<matplotlib.figure.Figure at 0x7faeff5e7190>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "import numpy as np\n",
+    "\n",
+    "x = np.arange(10)\n",
+    "x = (x*10) * 1681\n",
+    "plt.figure(figsize=(12,6))\n",
+    "plt.plot(x, pairwise_response_times)\n",
+    "plt.plot(x, Annoy_response_times[:10])\n",
+    "plt.plot(x, nms_response_times)\n",
+    "plt.title('Response time from nearest neighbours query')\n",
+    "plt.legend(['Pairwise dist.', 'Annoy', 'nms'], loc='upper left')\n",
+    "plt.xlabel('Number of vectors')\n",
+    "plt.ylabel('Avg. response time (ms)')\n",
+    "plt.yscale('log')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<h3>So, should you use one of these libraries?</h3>\n",
+    "\n",
+    "On a smaller dataset, it's not worth the hassle and loss of accuracy for a small increase in speed. However, if you need to respond in < 100ms, then your vector set can only grow so large before you'll need to start thinking about one of these libraries. \n",
+    "\n",
+    "Annoy has the advantage of saving to a memory map, and so playing well with multiple processes in a production environment\n",
+    "\n",
+    "NMSlib has generally been shown to outperform Annoy in terms of speed and accuracy, but I also found that constructing indexes over a large dataset was quite slow. This becomes a problem if you have to rebuild your index frequently because of fast moving product catalogues etc. "
```

Comment:
Maybe try without the {'post':2} option for NMSLib - it roughly doubles the time to create a nmslib index. It does improve the recall rate, but its still possible to get a good QPS at over a 99% recall without it.   I have some breakdowns comparing the times with and without it here: https://github.com/benfred/bens-blog-code/tree/master/approximate_als.

Also I think you might want to note that Annoy is a pretty good choice for calculating similar items, but for some reason it isn't a great choice for recommending items after making that transform. I benchmarked this using Annoy/NMSLib/Faiss for recommending items here: http://www.benfrederickson.com/approximate-nearest-neighbours-for-recommender-systems/  - and its not clear that Annoy is better than just using a brute force solution for recommending items.